src/stochprog.jl

```diff
@@ -0,0 +1,112 @@
+using LightGraphs
+
+export AbstractStochasticProgram, stochasticprogram
+export isleaf, probability, cutgenerator, numberofpaths
+export setprobability!, add_scenario_state!, add_scenario_transition!
+
+"""
+    AbstractStochasticProgram
+
+Stochastic program instance
+"""
+abstract type AbstractStochasticProgram <: AbstractGraph{Int} end
+
+"""
+    stochasticprogram(args...)
+
+Creates a stochastic program from the arguments
+"""
+function stochasticprogram end
+
+"""
+    getmaster(sp::AbstractStochasticProgram)
+
+Returns the master node of stochastic problem `sp`.
+"""
+function getmaster end
+
+"""
+    probability(sp::AbstractStochasticProgram, edge)
+
+Returns the probability to take the edge `edge` in the stochastic problem `sp`.
+"""
+function probability end
+
+"""
+    setchildx!(sp::AbstractStochasticProgram, node, child, sol)
+
+Sets the parent solution of `child` as `sol`, the solution obtained at `node`.
+"""
+function setchildx! end
+
+"""
+    solve!(sp::AbstractStochasticProgram, node)
+
+Solves the program at node `node` in `sp`.
+"""
+function solve! end
+
+"""
+    getobjlb(sp::AbstractStochasticProgram, node)
+
+Gets the current lower bound to the objective of `node`.
+"""
+function getobjlb end
+
+"""
+    setθlb!(sp::AbstractStochasticProgram, node, θlb)
+
+Sets the lower bounds to the objective of the children of `node` to `θlb`.
+"""
+function setθlb! end
+
+"""
+    statedim(sp::AbstractStochasticProgram, node)
+
+Returns the dimension of the state at `node`.
+"""
+function statedim end
```

Comment:
I prefer `statedim(sp)`. The state dimension should not change at different nodes in the graph

Reply:
In the [EntropicCone](https://github.com/blegat/EntropicCone.jl) application, it does :/

Reply:
What is a "state"? If we think about reservoirs, there might be 2 in the first stage, and 3 in the second. However, there really is 3 in every stage, it's just that in the first stage the 3rd reservoir has a volume of 0. State dimension is a constant in a stochastic program.

Reply:
When it has a physical meaning, I agree but in an abstract mathematical stochastic problem (it is the case with the entropic cone), there is not clear relation between the states from one node to the other one and the number change. In fact with the entropic cone, the graph is infinite and deep down in the graph, there are nodes of infinite number of states !

Reply:
This is because you're only using feasibility cuts right? I still don't fully understand your model haha

Reply:
I don't think it is related to feasibility cuts